File: src/ServerPulse.Backend/AnalyzerApi/Domain/Dtos/Responses/ServerLoadStatisticsResponse.cs
using AnalyzerApi.Domain.Dtos.Wrappers;

namespace AnalyzerApi.Domain.Dtos.Responses
{
    public class ServerLoadStatisticsResponse : BaseStatisticsResponse
    {
        public int AmountOfEvents { get; set; }
        public LoadEventWrapper? LastEvent { get; set; }
        public LoadMethodStatisticsResponse? LoadMethodStatistics { get; set; }
    }
}

File: src/ServerPulse.Frontend/src/app/modules/shared/domain/dtos/statistics/response/serverLoadStatisticsResponse.ts
import { convertToServerLoadResponse, convertToSLoadMethodStatisticsResponse, LoadMethodStatisticsResponse, ServerLoadResponse } from "../../../../index";

export interface ServerLoadStatisticsResponse {
    amountOfEvents: number;
    lastEvent: ServerLoadResponse | null;
    loadMethodStatistics: LoadMethodStatisticsResponse | null;
    collectedDateUTC: Date;
    isInitial: boolean;
}

export function convertToServerLoadStatisticsResponse(data: any): ServerLoadStatisticsResponse {
    return {
        amountOfEvents: data?.AmountOfEvents,
        lastEvent: convertToServerLoadResponse(data?.LastEvent),
        loadMethodStatistics: convertToSLoadMethodStatisticsResponse(data?.LoadMethodStatistics),
        collectedDateUTC: data?.CollectedDateUTC,
        isInitial: data?.IsInitial
    };
}